src/language-js/print/type-parameters.js

```diff
@@ -52,17 +52,22 @@ function printTypeParameters(path, options, print, paramsKey) {
     ];
   }
 
+  // Keep comma if the file extension is .tsx and
+  // has one type parameter that isn't extend with any types.
+  // Because, otherwise formatted result will be invalid as tsx.
+  const trailingComma =
+    getFunctionParameters(n).length === 1 &&
+    isTSXFile(options) &&
+    !n[paramsKey][0].constraint &&
+    path.getParentNode().type === "ArrowFunctionExpression"
+      ? ","
+      : ifBreak(shouldPrintComma(options, "all") ? "," : "");
```

Comment:
Please extract the `shouldPrintComma` condition out of `ifBreak` like I did in `print/array.js`:

https://github.com/prettier/prettier/blob/bba719283d484f27659e2a69ff214cf287f15ab7/src/language-js/print/array.js#L60-L66